.github/workflows/test-and-build.yml

```diff
@@ -0,0 +1,165 @@
+name: Test and Build
+on:
+  pull_request:
+    types:
+      - opened
+  push:
+
+jobs:
+  tests:
+    name: Lint and Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          show-progress: false
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12.3'
+          check-latest: true
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install pytest pytest-cov coveralls
+
+      - name: Run linting
+        run: |
+          python -m pylint src
+          python -m pylint tests
+      - name: Run tests with coverage
+        run: |
+          python -m pytest -v tests --cov=src --cov-report=term --cov-report=xml
+
+      - name: Send coverage to Coveralls
+        uses: coverallsapp/github-action@v2
+        with:
+          file: coverage.xml
+          flag-name: python-3.12.3
+          parallel: true
+
+  post-tests:
+    needs: tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Coveralls Finished
+        uses: coverallsapp/github-action@v2
+        with:
+          parallel-finished: true
+
+  build:
+    name: Build Docker Images
+    runs-on: ubuntu-latest
+    if: ${{ startsWith(github.ref, 'refs/tags/') || github.ref == format('refs/heads/{0}', github.event.repository.default_branch) }}
+    permissions:
+      actions: read
+      contents: write
+      packages: write
+      id-token: write
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      with:
+        show-progress: false
+
+    - name: Docker Login to GitHub Repository
+      uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567 # v3.3.0
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Docker Login to DockerHub
+      uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567 # v3.3.0
+      with:
+        username: ${{ secrets.BOT_DOCKER_USERNAME }}
+        password: ${{ secrets.BOT_DOCKER_TOKEN }}
+
+    - name: Login to Mia registry
+      uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567 # v3.3.0
+      with:
+        registry: nexus.mia-platform.eu
+        username: ${{ secrets.MIA_PLATFORM_DOCKER_REGISTRY_USERNAME }}
```

Comment:
these secrets have a different name in this organization. See crud-service actions